go/pvl/helpers.go

```diff
@@ -130,17 +130,26 @@ func jsonStringSimple(object *jsonw.Wrapper) (string, error) {
 	return "", fmt.Errorf("Non-simple object: %v", object)
 }
 
-// selectionContents gets the HTML contents of all elements in a selection, concatenated by a space.
+// selectionContents gets the value of all elements in a selection, concatenated by a space.
 // If getting the contents/attr value of any elements fails, that does not cause an error.
+// The value of an element can be:
+// 1. its Text (if attr=="" and data==false)
+// 2. its Data (if attr=="" and data==true)
+// 3. an attribute (if attr=="something" and data==false)
+// The value of an element is its Text, unless one of
 // The result can be an empty string.
-func selectionContents(selection *goquery.Selection, useAttr bool, attr string) string {
+func selectionContents(selection *goquery.Selection, attr string, data bool) string {
```

Comment:
Split into functions `selectionText`, `selectionAttr`, and `selectionData`. Yes you're right the contents word got overloaded, these names match goquery.

Check out the callsite, because it might still bother you. Basically because `selector_css` has the same problem, but I think it would be weird to split it up the same way.